ArchLinux/Dockerfile

```diff
@@ -1,41 +1,58 @@
 FROM base/archlinux:latest
-MAINTAINER Philipp Moeller <bootsarehax@gmail.com>
+MAINTAINER Philipp Moeller <bootsarehax@gmail.com>, Alexander Kobel <alexander.kobel@mpi-inf.mpg.de>
 
 # base/archlinux is really old; in particular, the packager keys are horribly outdated.
-# We delete the original keyring and refresh it from scratch before updating and installing new packages.
-RUN pacman -Sy --noconfirm && \
-    rm -rf /etc/pacman.d/gnupg && \
-    pacman-key --init && \
-    pacman-key --populate archlinux && \
-    pacman-key --refresh-keys && \
-    pacman -Su --noconfirm && \
+# We delete the original keyring and refresh it from scratch before the real setup:
+
+# install the official Arch Linux keyring package
+RUN pacman -Sy --noconfirm archlinux-keyring
+
+# remove the old database of trusted keys the hard way and initialize a new database
+# (Commented because it is currently unnecessary and takes a while;
+# should only be used as a last resort if the base image is /really, really/ old.)
+RUN rm -rf /etc/pacman.d/gnupg && \
+    pacman-key --init
```

Comment:
Whoopsie. Indeed; it turned out to be necessary, but only a few lines after that. I updated the comment.